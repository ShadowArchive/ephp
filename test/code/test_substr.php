<?php

print "substr([], 0);\n";
var_dump(substr([], 0));

print "substr(\"abcdef\", 0, 0);\n";
var_dump(substr("abcdef", 0, 0));

print "substr(\"abcdef\", -1);\n";
var_dump(substr("abcdef", -1));
print "substr(\"abcdef\", -2);\n";
var_dump(substr("abcdef", -2));
print "substr(\"abcdef\", -7);\n";
var_dump(substr("abcdef", -7));
print "substr(\"abcdef\", -3, 1);\n";
var_dump(substr("abcdef", -3, 1));
print "substr(\"abcdef\", -3, 3);\n";
var_dump(substr("abcdef", -3, 3));
print "substr(\"abcdef\", -3, 4);\n";
var_dump(substr("abcdef", -3, 4));
print "substr(\"abcdef\", -3, -1);\n";
var_dump(substr("abcdef", -3, -1));
print "substr(\"abcdef\", -3, -3);\n";
var_dump(substr("abcdef", -3, -3));
print "substr(\"abcdef\", -3, -4);\n";
var_dump(substr("abcdef", -3, -4));

print "substr(\"abcdef\", 0);\n";
var_dump(substr("abcdef", 0));
print "substr(\"abcdef\", 1);\n";
var_dump(substr("abcdef", 1));
print "substr(\"abcdef\", 6);\n";
var_dump(substr("abcdef", 6));
print "substr(\"abcdef\", 3, 1);\n";
var_dump(substr("abcdef", 3, 1));
print "substr(\"abcdef\", 3, 3);\n";
var_dump(substr("abcdef", 3, 3));
print "substr(\"abcdef\", 3, 4);\n";
var_dump(substr("abcdef", 3, 4));
print "substr(\"abcdef\", 3, -1);\n";
var_dump(substr("abcdef", 3, -1));
print "substr(\"abcdef\", 3, -3);\n";
var_dump(substr("abcdef", 3, -3));
print "substr(\"abcdef\", 3, -4);\n";
var_dump(substr("abcdef", 3, -4));
