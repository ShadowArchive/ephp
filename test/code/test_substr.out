substr([], 0);

Warning: substr() expects parameter 1 to be string, array given in {{CWD}}/test/code/test_substr.php on line 4
NULL
substr("abcdef", 0, 0);
string(0) ""
substr("abcdef", -1);
string(1) "f"
substr("abcdef", -2);
string(2) "ef"
substr("abcdef", -7);
string(6) "abcdef"
substr("abcdef", -3, 1);
string(1) "d"
substr("abcdef", -3, 3);
string(3) "def"
substr("abcdef", -3, 4);
string(3) "def"
substr("abcdef", -3, -1);
string(2) "de"
substr("abcdef", -3, -3);
string(0) ""
substr("abcdef", -3, -4);
string(0) ""
substr("abcdef", 0);
string(6) "abcdef"
substr("abcdef", 1);
string(5) "bcdef"
substr("abcdef", 6);
string(6) "abcdef"
substr("abcdef", 3, 1);
string(1) "d"
substr("abcdef", 3, 3);
string(3) "def"
substr("abcdef", 3, 4);
string(3) "def"
substr("abcdef", 3, -1);
string(2) "de"
substr("abcdef", 3, -3);
string(0) ""
substr("abcdef", 3, -4);
string(0) ""
